### ISSUE TITLE
Instaloader stuck at downloading comments

### ISSUE THREAD

**Issue by drzazga3009** [instaloader/instaloader#1921]
**Describe the bug**
After some time instaloader gets stuck, just printing numbers like `250...500...750...` and so on.
It seems this is generated in this line https://github.com/instaloader/instaloader/blob/master/instaloader/instaloader.py#L407
and is related to downloading comments.
The problem is that the Instagram profile I archive has at max around 250 comments per post. In one case instaloader got up to 50000 before I noticed it got stuck. I checked the exact post and it has 157 comments. 

**To Reproduce**
`instaloader --login <username> --sessionfile="file.dat" --stories --highlights --comments --geotags --no-compress-json --filename-pattern="{shortcode}/{date_utc}_UTC" soobly_s2 dayomi99 yeolum_e e_lludda`
One of the posts for which it got stuck is https://www.instagram.com/p/Cgjb7d-rgYP/
file.dat is a session file generated with https://raw.githubusercontent.com/instaloader/instaloader/master/docs/codesnippets/615_import_firefox_session.py
Without this file instaloader is unable to download the highest res pictures after a few posts, it prints error 401 and fallbacks to 1350px.

**Expected behavior**
Just download all comments and move one, it works for some posts. Alternatively, allow defining a maximum number of comments to download.

**Error messages and tracebacks**
No error messages, just integers divisible by 250, separated with '...'.

**Instaloader version**
```
$ instaloader --version
4.9.6
```

**Comment by drzazga3009** [comment]
After playing a little bit with it I noticed this could be related to API responses when fetching comments. In fact, no response was printed so I assume it was a 2xx code response but after some time the session was invalidated and I started getting 401 when downloading photos. Even though I imported the session from Firefox, instaloader was downloading at max 1350px images when 1800px are available.

When I run instaloader without `--comments` everything seems to work correctly, I didn't get 401 error after downloading ~200 posts.

**Comment by aandergr** [comment]
Downloading comments seems to work for me. Is there any specific post, where downloading comments does not work properly, and could you share its shortcode?

**Comment by drzazga3009** [comment]
Thanks for the reply. `Cgjb7d-rgYP` got deleted but iirc there was no specific post that made instaloader hang, it just happened after ~tens of saved posts. Eventually, I gave up on saving comments and was able to archive all profiles I wanted. After ~2 weeks of using instaloader in `--fast-update` mode  I gave up because it gets my account locked immediately no matter what settings I use.

Hope someone finds fixes for this problem and the login problem, I just gave up since I archived all older posts and now just save new ones by hand.
